app/notifications/process_notifications.py

```diff
@@ -295,70 +295,76 @@ def send_notification_to_queue(notification, research_mode, queue=None):
-        service_id = notification.get("service").id if notification.get("service") else None  # type: ignore
+    for verifiedNotification in verifiedNotifications:
+        notification_created_at = verifiedNotification.created_at or datetime.utcnow()
+        notification_id = verifiedNotification.notification_id or uuid.uuid4()
+        notification_recipient = verifiedNotification.recipient or verifiedNotification.to
+        service_id = verifiedNotification.service.id if verifiedNotification.service else None  # type: ignore
         # todo: potential bug. notification_obj is being created using some keys that don't exist on notification
         # reference, created_by_id, status, billable_units aren't keys on notification at this point
         notification_obj = Notification(
             id=notification_id,
-            template_id=notification.get("template_id"),
-            template_version=notification.get("template_version"),
+            template_id=verifiedNotification.template_id,
+            template_version=verifiedNotification.template_version,
             to=notification_recipient,
             service_id=service_id,
-            personalisation=notification.get("personalisation"),
-            notification_type=notification.get("notification_type"),
-            api_key_id=notification.get("api_key_id"),
-            key_type=notification.get("key_type"),
+            personalisation=verifiedNotification.personalisation,
+            notification_type=verifiedNotification.notification_type,
+            api_key_id=verifiedNotification.api_key_id,
+            key_type=verifiedNotification.key_type,
             created_at=notification_created_at,
-            job_id=notification.get("job_id"),
-            job_row_number=notification.get("job_row_number"),
-            client_reference=notification.get("client_reference"),
-            reference=notification.get("reference"),  # type: ignore
-            created_by_id=notification.get("created_by_id"),  # type: ignore
-            status=notification.get("status"),  # type: ignore
-            reply_to_text=notification.get("reply_to_text"),
-            billable_units=notification.get("billable_units"),  # type: ignore
+            job_id=verifiedNotification.job_id,
+            job_row_number=verifiedNotification.job_row_number,
+            client_reference=verifiedNotification.client_reference,
+            # REVIEW: We can remove these ones if possible, as these will be set later in the process:
+            #   reference: this is the provider's reference and will be set on sending time
+            #   created_by_id: this is the user who created the notification and will be set on sending time, used by one off or admin UI uploads
+            # reference=verifiedNotification.reference,  # type: ignore
+            # created_by_id=verifiedNotification.created_by_id,  # type: ignore
+            # billable_units=verifiedNotification.billable_units,  # type: ignore
```

Comment:
I think we can remove these:
* `reference`: a notification will only have a reference once SES or Pinpoint is called. The exception is letters where a random reference will be generated but this function does not handle these, not GCNotify at large. 
* `created_by_id`: Not set when received by the API AFAIK. Other code paths are used by the one-off send or bulk send, where the `created_by_id` might be set.
* `billable_units`: most likely not set at this stage yet.